layouts/index.html

```diff
@@ -7,15 +7,24 @@
                     {{ .Content }}
                 </div>
             {{ end }}
-            <div class="page-heading">Blog</div>
-            <ul>
-            {{ range .Data.Pages.ByPublishDate }}
-                {{ if eq .Section "blog" }}
-                    {{ partial "li.html" . }}
-                {{ end }}
+            {{ $totalpostscount := len (where .Data.Pages "Section" "blog") }}
+            {{ $latestpostscount := .Site.Params.latestpostscount | default $totalpostscount }}
+            {{ if gt $latestpostscount 0 }}
+                <div class="page-heading">Latest posts</div>
+                <ul>
+                    {{ range (first $latestpostscount (where .Data.Pages.ByPublishDate "Section" "blog")) }}
+                        {{ partial "li.html" . }}
+                    {{ end }}
+                    {{ if gt $totalpostscount $latestpostscount }}
+                        {{ range where .Site.Menus.main "Identifier" "blog" }}
+                            <li class="post-item no-bullet">
+                                <span class="meta"></span> <a {{ printf "href=%q " .URL | safeHTMLAttr }}><span>See more&hellip;</span></a>
```

Comment:
Add `id="see-more"` for the See More link. And style it with `margin-top: 1rem;` in `site.less` for visual separation?